src/generate-query.ts

```diff
@@ -509,9 +506,52 @@ function getMissingSlicingArg(
   if (usesSlicingArg) return null
 
   // Return the first slicing arguments:
-  return field.arguments.find((arg) =>
-    slicingArguments.includes(arg.name.value)
+  return field.arguments.find((arg) => {
+    return slicingArguments.find((slicingArgument) => {
+      let ret = getMissingSlicingArgHelper(
+        slicingArgument.split('.'),
+        arg,
+        schema
+      )
+      if (ret) return arg
```

Comment:
This is strange to me. The callback function inside a `find()` iterator should return a boolean but you are return `arg`?

Reply:
fixed